src/multi_physics/QED/src/math/vec_functions.hpp

```diff
@@ -0,0 +1,182 @@
+#ifndef PICSAR_MULTIPHYSICS_VEC_FUNCTIONS
+#define PICSAR_MULTIPHYSICS_VEC_FUNCTIONS
+
+//Should be included by all the src files of the library
+#include "../qed_commons.h"
+
+//Uses GPU-friendly arrays
+#include "../containers/picsar_array.hpp"
+//Uses sqrt
+#include "cmath_overloads.hpp"
+
+namespace picsar{
+namespace multi_physics{
+namespace math{
+
+    //This .hpp file contains functions to perform operations on 3-vectors
+    // (norm, scalar multiplication, vector and cross product...)
+    template <typename RealType>
+    using vec3 = containers::picsar_array<RealType, 3>;
+
+    /**
+    * This function returns the squared norm of a 3-vector
+    *
+    * @tparam RealType the floating point type to be used
+    * @param[in] the vector for which the squared norm has to be calculated
+    * @return the squared norm
+    */
+    template <typename RealType>
+    PXRMP_INTERNAL_GPU_DECORATOR PXRMP_INTERNAL_FORCE_INLINE_DECORATOR
+    RealType norm2(const vec3<RealType>& vec) noexcept
```

Comment:
Maybe rename `norm_square` to avoid confusion with the usual `norm2` which is the square root of that?

Reply:
I agree with you, norm2 can be quite confusing! I've just changed it...